Core/Inc/controls/pid.h

```diff
@@ -0,0 +1,172 @@
+/**
+ ******************************************************************************
+ * @file    pid.h
+ * @brief   PID Controller Library for Drone Flight Control
+ * @author  Ulysses Flight Controller
+ * @date    2025-11-23
+ ******************************************************************************
+ * @attention
+ *
+ * This PID controller is designed for drone rate control (roll, pitch, yaw).
+ * It implements:
+ *   - Derivative on measurement (prevents derivative kick on setpoint changes)
+ *   - Integral anti-windup (prevents integral term overflow)
+ *   - Output limiting (constrains control output)
+ *
+ * Usage Example:
+ * @code
+ * // Initialize PID controller for roll axis
+ * PID_Controller_t rollPID;
+ * PID_Init(&rollPID, 1.5f, 0.5f, 0.05f, 100.0f, -500.0f, 500.0f);
+ *
+ * // In your main flight control loop:
+ * float dt = 0.002f;  // 2ms loop time (500Hz)
+ * float desiredRollRate = 45.0f;  // deg/s from pilot stick input
+ * float currentGyroRate = gyro.x;  // deg/s from IMU
+ *
+ * float motorCorrection = PID_Compute(&rollPID, desiredRollRate, currentGyroRate, dt);
+ * @endcode
+ *
+ ******************************************************************************
+ */
+
+#ifndef PID_H
+#define PID_H
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/* Includes ------------------------------------------------------------------*/
+#include <stdint.h>
+
+/* Exported types ------------------------------------------------------------*/
+
+/**
+ * @brief PID Controller structure
+ *
+ * This structure holds all state and configuration for a single PID controller.
+ * For drone control, you'll typically need three instances: roll, pitch, yaw.
+ */
+typedef struct {
+    /* Tuning gains */
+    float Kp;                   /**< Proportional gain */
+    float Ki;                   /**< Integral gain */
+    float Kd;                   /**< Derivative gain */
+
+    /* Internal state (do not modify directly) */
+    float integralSum;          /**< Accumulated integral term */

```

Comment:
nit: for consistency, can we use snake case for all symbols
```suggestion
    float integral_sum;          /**< Accumulated integral term */
```